File: LibraryManagementProject/src/main/java/com/libraryManagementSystem/service/FineService.java
package com.libraryManagementSystem.service;

import com.libraryManagementSystem.dto.FineDto;
import com.libraryManagementSystem.dto.PayFineRequest;

import java.util.List;

public interface FineService {
    FineDto payFine(PayFineRequest request);
    FineDto waiveFine(Long fineId);
    List<FineDto> getAllFines();
    List<FineDto> getUserFines(Long userId);
}

File: LibraryManagementProject/frontend/src/services/fineService.ts
import API from './api'
import type { FineDto } from '../types'

const BASE = '/fines'

const fineService = {
  /** Get all fines (admin/librarian) */
  getAllFines: () =>
    API.get<FineDto[]>(BASE).then((r) => r.data),

  /** Get fines for a specific user */
  getUserFines: (userId: number) =>
    API.get<FineDto[]>(`${BASE}/user/${userId}`).then((r) => r.data),

  /** Mark a fine as paid */
  payFine: (fineId: number) =>
    API.post<FineDto>(`${BASE}/${fineId}/pay`).then((r) => r.data),
}

export default fineService
